CruCentralCoast/Community/CommunityGroups/CommunityGroupDetailsVC.swift

```diff
@@ -7,41 +7,87 @@
         super.viewDidLoad()
 
         self.navigationItem.largeTitleDisplayMode = .never
+        
     }
     
-    @IBAction func joinCommunityGroup() {
-        // TODO
-        self.presentAlert(title: "Join Community Group", message: "Coming Soon...")
+    @IBAction func didTapContactLeader() {
+        if MFMessageComposeViewController.canSendText() {
+            if self.leaderPhoneNumbers.isEmpty {
+                self.presentAlert(title: "Can't Contact Leader", message: "Sorry, there is no phone number listed for this group")
+            }
+            
+            else {
+                let controller = MFMessageComposeViewController()
+                controller.body = "Hey I'm interested in joining your community group!"
+                controller.recipients = self.leaderPhoneNumbers
+                controller.messageComposeDelegate = self
+                self.present(controller, animated: true, completion: nil)
+            }
+            
+        }
+        else {
+            print("error cant send text")
+        }
+    }
+    
+    func messageComposeViewController(_ controller: MFMessageComposeViewController, didFinishWith result: MessageComposeResult) {
+        controller.dismiss(animated: true, completion: nil)
     }
     
+    
     func configure(with communityGroup: CommunityGroup) {
         DispatchQueue.main.async {
-            self.genderLabel.text = communityGroup.gender.rawValue.uppercased()
-            self.dayLabel.text = communityGroup.weekDay.rawValue.uppercased()
-            self.timeLabel.text = communityGroup.time
-            self.nameLabel.text = communityGroup.name
-            self.yearLabel.text = communityGroup.year.rawValue.uppercased()
+            
+            var leaderArray : [String] = []
+            for leader in communityGroup.leaders {
+                leaderArray.append(leader.name)
+            }
+            self.nameLabel.text = leaderArray.joined(separator: ", ")
+            
             self.movementLabel.text = communityGroup.movement?.name
-            self.leaderNamesLabel.text = "Leaders: \(communityGroup.leaderNames ?? "N/A")"
-            self.summaryLabel.text = communityGroup.summary
+            
+            let gender = communityGroup.gender.rawValue.localizedCapitalized
+            let day = communityGroup.weekDay.rawValue.localizedCapitalized
+            let time = communityGroup.time ?? "N/A"
+            let year = communityGroup.year.rawValue.localizedCapitalized 
+            
+            if day == "" || time == "" {
+                self.dayTimeLabel.text = "No meeting time listed"
+            }
+            else {
+                self.dayTimeLabel.text = "Meets on " + day + " at " + time
+            }
+            
+            self.yearGenderLabel.text = year + " | " + gender
+            
+            //self.leaderNamesLabel.text = "Leaders: \(communityGroup.leaderNames ?? "N/A")"
+            
+            for leader in communityGroup.leaders {
```

Comment:
I think you could reduce this to something like `communityGroup.leaders.filter({ $0.phone != nil }).compactMap($0.phone!)`